prow/plugins/lifecycle/lifecycle_test.go

```diff
@@ -17,133 +17,248 @@ limitations under the License.
-	c.removed = append(c.removed, label)
-	return nil
+func formatLabels(labels ...string) []string {
+	r := []string{}
+	for _, l := range labels {
+		r = append(r, fmt.Sprintf("%s/%s#%d:%s", "org", "repo", 1, l))
+	}
+	if len(r) == 0 {
+		return nil
+	}
+	return r
 }
 
 func TestDeprecatedClose(t *testing.T) {
-	fc := &fakeClient{}
-	gce := &github.GenericCommentEvent{}
+	fc := &fakegithub.FakeClient{
+		Issues:        make([]github.Issue, 1),
+		IssueComments: make(map[int][]github.IssueComment),
+	}
+	gce := &github.GenericCommentEvent{
+		Action: github.GenericCommentActionCreated,
+		Number: 1,
+		Repo:   github.Repo{Owner: github.User{Login: "org"}, Name: "repo"},
+		User:   github.User{Login: "Alice"},
+	}
 	ticker := make(chan time.Time, 1)
 	deprecatedTick = ticker
 	err := deprecate(fc, "fake", "org", "repo", 1, gce)
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if fc.commented {
+	if len(fc.IssueCommentsAdded) != 0 {
 		t.Fatalf("should not comment on empty ticker")
 	}
 	ticker <- time.Now()
 	err = deprecate(fc, "fake", "org", "repo", 1, gce)
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
-	if !fc.commented {
+	if len(fc.IssueCommentsAdded) == 0 {
 		t.Fatalf("must comment on filled timer")
 	}
 
 }
 
 func TestAddLifecycleLabels(t *testing.T) {
 	var testcases = []struct {
-		body    string
-		added   []string
-		removed []string
-		comment bool
+		name                  string
+		body                  string
+		expectedNewLabels     []string
+		expectedRemovedLabels []string
+		issueLabels           []string
 	}{
 		{
-			body: "/random-command",
+			name:                  "random command",
+			body:                  "/random-command",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
+		},
+		{
+			name:                  "remove lifecycle, but don't specify which state",
+			body:                  "/remove-lifecycle",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
+		},
+		{
+			name:                  "add lifecycle, but don't specify which state",
+			body:                  "/lifecycle",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
+		},
+		{
+			name:                  "add lifecycle random",
+			body:                  "/lifecycle random",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body: "/remove-lifecycle",
+			name:                  "remove lifecycle random",
+			body:                  "/remove-lifecycle random",
+			expectedNewLabels:     []string{},
+			expectedRemovedLabels: []string{},
+			issueLabels:           []string{},
 		},
 		{
-			body: "/lifecycle",
+			name:                  "add lifecycle frozen, don't have it",
```

Comment:
Consider making the name also include the expected result just not the setup (we're trying to test here that we will add a frozen label when it isn't already present)

Reply:
> Consider making the name also include the expected result

Done. Ptal.